src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,203 @@
+                {
+                    lockFileTargetLibrary.Type = reader.ReadNextTokenAsString();
+                }
+                else if (reader.ValueTextEquals(FrameworkPropertyName))
+                {
+                    lockFileTargetLibrary.Framework = reader.ReadNextTokenAsString();
+                }
+                else if (reader.ValueTextEquals(DependenciesPropertyName))
+                {
+                    reader.Read();
+                    if (ReadPackageDependencyList(ref reader) is { Count: not 0 } packageDependencies)
+                    {
+                        lockFileTargetLibrary.Dependencies = packageDependencies;
+                    }
+                }
+                else if (reader.ValueTextEquals(FrameworkAssembliesPropertyName))
+                {
+                    reader.Read();
+                    lockFileTargetLibrary.FrameworkAssemblies = reader.ReadStringArrayAsIList(new List<string>());
+                }
+                else if (reader.ValueTextEquals(RuntimePropertyName))
+                {
+                    reader.Read();
+                    if (reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter) is { Count: not 0 } runtimeAssemblies)
+                    {
+                        lockFileTargetLibrary.RuntimeAssemblies = runtimeAssemblies;
+                    }
+                }
+                else if (reader.ValueTextEquals(CompilePropertyName))
+                {
+                    reader.Read();
+                    if (reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter) is { Count: not 0 } compileTimeAssemblies)
+                    {
+                        lockFileTargetLibrary.CompileTimeAssemblies = compileTimeAssemblies;
+                    }
+                }
+                else if (reader.ValueTextEquals(ResourcePropertyName))
+                {
+                    reader.Read();
+                    if (reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter) is { Count: not 0 } resourceAssemblies)
+                    {
+                        lockFileTargetLibrary.ResourceAssemblies = resourceAssemblies;
+                    }
+                }
+                else if (reader.ValueTextEquals(NativePropertyName))
+                {
+                    reader.Read();
+                    if (reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter) is { Count: not 0 } nativeLibraries)
+                    {
+                        lockFileTargetLibrary.NativeLibraries = nativeLibraries;
+                    }
+                }
+                else if (reader.ValueTextEquals(BuildPropertyName))
+                {
+                    reader.Read();
+                    if (reader.ReadObjectAsList<LockFileItem>(Utf8JsonReaderExtensions.LockFileItemConverter) is { Count: not 0 } build)
```

Comment:
Applies to all of the collections, are we better off setting Array.Empty instead of leaving it unset?
What happens if we leave it unset? Any potential issues with Array.Empty?

Reply:
I'll have to check through other instances of this but for the LockFileTargetLibrary the list is set in the getter for the property.

Reply:
It sets it if it's not there. 
If you already know you can have array empty, then I think you can just set that.

Reply:
Seems fine with me, updated.